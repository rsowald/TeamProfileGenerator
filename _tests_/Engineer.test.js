const Engineer = require('../lib/Engineer');

describe('Engineer class', () => {


    it('constructor and getter methods', () => {
        const engineer = new Engineer('John Doe', 458, 'doe458@fakemail.com', 'https://github.com/jdozer');
        expect(engineer.getName()).toEqual('John Doe');
        expect(engineer.getId()).toBe(458);
        expect(engineer.getEmail()).toBe('doe458@fakemail.com');
        expect(engineer.getGitHub()).toBe('https://github.com/jdozer');
        expect(engineer.getRole()).toBe('Engineer');
    });



    it('engineer card', () => {
        const engineer = new Engineer('John Doe', 458, 'doe458@fakemail.com', 'https://github.com/jdozer');
        expect(engineer.getEmployeeCard()).toMatchSnapshot();
    });
})