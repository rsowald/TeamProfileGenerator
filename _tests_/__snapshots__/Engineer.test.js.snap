// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engineer class engineer card 1`] = `
"<div class=\\"col mb-4\\">
    <div class=\\"card h-100\\">
        <div class=\\"card-header\\">
            <h4>Engineer<i class=\\"fas fa-glasses\\"></i></h4>
            <h4>John Doe</h4>
        </div>
        <div class=\\"card-body\\">
            <ul class=\\"list-group list-group-flush\\">
                <li class=\\"list-group-item\\">ID: 458</li>
                <li class=\\"list-group-item\\">Email: doe458@fakemail.com</li>
                <li class=\\"list-group-item\\">GitHub: https://github.com/jdozer</li>
            </ul>
        </div>
    </div>
    </div>"
`;
